libr/include/r_util/r_str.h

```diff
@@ -23,12 +24,38 @@ typedef int (*RStrRangeCallback) (void *, int);
 
 static inline void r_str_rmch(char *s, char ch) {
 	for (;*s; s++) {
-		if (*s==ch) {
+		if (*s == ch) {
 			memmove (s, s + 1, strlen (s));
 		}
 	}
 }
 
+//  -- strf
+typedef struct r_strf_t {
+	size_t idx, size;
+	char buf[0];
+} r_strf__;
+
+R_NULLABLE static inline char *r_strf_(r_strf__ *s, const char *fmt, ...) {
+	r_return_val_if_fail (s && fmt, NULL);
+	if (s->idx >= s->size) {
+		s->idx = 0;
+	}
+	va_list ap;
+	va_start (ap, fmt);
+	char *p = s->buf + s->idx;
+	size_t left = s->size - s->idx - 1;
+	int res = vsnprintf (p, left, fmt, ap);
+	r_return_val_if_fail (res != left, NULL);
```

Comment:
As said in another comment, this is wrong. It should probably be `res < left`

Reply:
ths is fixed